projects/epc/playground/src/groups/HardwareSourcesGroup.cpp

```diff
@@ -12,69 +12,74 @@ HardwareSourcesGroup::HardwareSourcesGroup(ParameterGroupSet* parent, Settings*
+                                                       ScaleConverter::get<LinearBipolar100PercentScaleConverter>()));
 
   appendParameter(new HardwareSourceSendParameter(this, *bender, getBenderSendID(),
-                                                  ScaleConverter::get<LinearBipolar100PercentScaleConverter>(), 0, 100,
-                                                  1000, m_settings));
+                                                  ScaleConverter::get<LinearBipolar100PercentScaleConverter>()));
 
-  auto at = appendParameter(new AftertouchParameter(this, getAftertouchParameterID(),
-                                          ScaleConverter::get<Linear100PercentScaleConverter>(), 0, 100, 1000));
+  auto at = appendParameter(
+      new AftertouchParameter(this, getAftertouchParameterID(), ScaleConverter::get<Linear100PercentScaleConverter>()));
 
   appendParameter(new HardwareSourceSendParameter(this, *at, getAftertouchSendID(),
-                                                  ScaleConverter::get<Linear100PercentScaleConverter>(), 0, 100, 1000,
-                                                  m_settings));
+                                                  ScaleConverter::get<Linear100PercentScaleConverter>()));
 
-  auto ribbon1 = appendParameter(new RibbonParameter(this, getUpperRibbonParameterID(),
-                                      ScaleConverter::get<Linear100PercentScaleConverter>(), 0.5, 100, 1000));
+  auto ribbon1 = appendParameter(
+      new RibbonParameter(this, getUpperRibbonParameterID(), ScaleConverter::get<Linear100PercentScaleConverter>()));
 
   appendParameter(new HardwareSourceSendParameter(this, *ribbon1, getRibbon1SendID(),
-                                                  ScaleConverter::get<Linear100PercentScaleConverter>(), 0.5, 100, 1000,
-                                                  m_settings));
+                                                  ScaleConverter::get<Linear100PercentScaleConverter>()));
 
-  auto ribbon2 = appendParameter(new RibbonParameter(this, getLowerRibbonParameterID(),
-                                      ScaleConverter::get<Linear100PercentScaleConverter>(), 0.5, 100, 1000));
+  auto ribbon2 = appendParameter(
+      new RibbonParameter(this, getLowerRibbonParameterID(), ScaleConverter::get<Linear100PercentScaleConverter>()));
 
   appendParameter(new HardwareSourceSendParameter(this, *ribbon2, getRibbon2SendID(),
-                                                  ScaleConverter::get<Linear100PercentScaleConverter>(), 0.5, 100, 1000,
-                                                  m_settings));
+                                                  ScaleConverter::get<Linear100PercentScaleConverter>()));
+}
+
+HardwareSourcesGroup::~HardwareSourcesGroup()
+{
+}
+
+void HardwareSourcesGroup::init()
+{
+  for(auto pedalID : { getPedal1ParameterID(), getPedal2ParameterID(), getPedal3ParameterID(), getPedal4ParameterID() })
```

Comment:
I think this could be written more easy to read and less complicated to executed, e.g.:

```suggestion
findAndCastParameterByID<PedalParameter>(getPedal1ParameterID())->init(settings.getSetting("Pedal1Type"));
findAndCastParameterByID<PedalParameter>(getPedal2ParameterID())->init(settings.getSetting("Pedal2Type"));
findAndCastParameterByID<PedalParameter>(getPedal3ParameterID())->init(settings.getSetting("Pedal3Type"));
findAndCastParameterByID<PedalParameter>(getPedal4ParameterID())->init(settings.getSetting("Pedal4Type"));
```


You would not only save a loops and branches here, but also in PedalParameter.